internal/util/fscrypt/fscrypt.go

```diff
@@ -217,12 +241,14 @@ func initializeAndUnlock(
 	}
 
 	protector, err := fscryptactions.CreateProtector(fscryptContext, protectorName, keyFn, owner)
-	if err != nil {
+	if err != nil && protector != nil {
```

Comment:
i looked at the CreateProtector it will always return the protector as nil when there is an error, we don't need a check for the protector if there is an error as nothing is to be reverted for that protector we can remove below code `protector.Revert()`  as well